libs/parallelism/execution/include/hpx/execution/algorithms/on.hpp

```diff
@@ -84,12 +84,20 @@ namespace hpx { namespace execution { namespace experimental {
             static constexpr bool sends_done = false;
 
             template <typename R>
-            auto connect(R&& r)
+            auto connect(R&& r) &&
             {
                 return hpx::execution::experimental::connect(std::move(s),
                     on_receiver<R, Scheduler>(
                         std::forward<R>(r), std::move(scheduler)));
             }
+
+            template <typename R>
+            auto connect(R&& r) &
+            {
+                return hpx::execution::experimental::connect(s,
+                    on_receiver<R, Scheduler>(
+                        std::forward<R>(r), std::move(scheduler)));
```

Comment:
Note to self, the scheduler should not be moved. Also check other `connect`s for similar mistakes.